raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1601,7 +1618,12 @@ private long pollFollower(long currentTimeMs) throws IOException {
     }
 
     private long pollFollowerAsVoter(FollowerState state, long currentTimeMs) throws IOException {
-        if (state.hasFetchTimeoutExpired(currentTimeMs)) {
+        GracefulShutdown shutdown = this.shutdown.get();
+        if (shutdown != null) {
+            // If we are a follower, then we can shutdown immediately. We want to
+            // skip the transition to candidate in any case.
+            return 0;
```

Comment:
Why the behavior of `pollFollowerAsVoter` and `pollVoted` are different when shutting down? Could the former case still help in casting and completing a vote as well?

Reply:
If we are a follower, then there is no election in progress to help with.